code/datums/status_effects/buffs.dm

```diff
@@ -262,3 +262,113 @@
 	deathrattle()
 	. = ..()
 
+/datum/status_effect/blooddrunk
+	id = "blooddrunk"
+	duration = 10
+	tick_interval = 0
+	alert_type = /obj/screen/alert/status_effect/blooddrunk
+	var/last_health = 0
+	var/last_bruteloss = 0
+	var/last_fireloss = 0
+	var/last_toxloss = 0
+	var/last_oxyloss = 0
+	var/last_cloneloss = 0
+	var/last_staminaloss = 0
+
+/obj/screen/alert/status_effect/blooddrunk
+	name = "Blood-Drunk"
+	desc = "You are drunk on blood! Your pulse thunders in your ears! Nothing can harm you!" //not true, and the item description mentions its actual effect
+	icon_state = "blooddrunk"
+
+/datum/status_effect/blooddrunk/on_apply()
+	. = ..()
+	if(.)
+		owner.maxHealth *= 10
+		owner.bruteloss *= 10
+		owner.fireloss *= 10
+		owner.toxloss *= 10
+		owner.oxyloss *= 10
+		owner.cloneloss *= 10
+		owner.staminaloss *= 10
+		owner.updatehealth()
+		last_health = owner.health
+		last_bruteloss = owner.getBruteLoss()
+		last_fireloss = owner.getFireLoss()
+		last_toxloss = owner.getToxLoss()
+		last_oxyloss = owner.getOxyLoss()
+		last_cloneloss = owner.getCloneLoss()
+		last_staminaloss = owner.getStaminaLoss()
+		owner.log_message("gained blood-drunk stun stun immunity", INDIVIDUAL_ATTACK_LOG)
```

Comment:
stun stun immunity: For when you're afraid the first person to doublestun will win the match...